### PR TITLE
tty: implement readv

### PR DESCRIPTION
**Pull request by petershh**

### CODE REVIEW

**Comment by petershh** [comment]
Should these commits be squashed?

File: kernel/kernel/tty/tty.cpp [hunk 1]
```diff
@@ -297,6 +297,29 @@ size_t ttydevfs_write(size_t offset, size_t len, void *ubuffer, struct file *f)
...
+    }
+
+    ssize_t copied = copy_from_iter(iter, buffer, iter->bytes);
+    if (copied < 0)
+    {
+        free(buffer);
+        return -EFAULT;
+    }
+
+    tty_write(buffer, copied, tty);
```
Comment (petershh) [inline kernel/kernel/tty/tty.cpp]: I think it is possible to make tty_write_iter function, so we can avoid allocating additional buffer

Comment (heatd) [reply kernel/kernel/tty/tty.cpp]: Yes, this is required, allocating a buffer is 1) too slow and 2) a possible DoS point given it takes a 64-bit user-controlled integer

**Comment by heatd** [comment]
Sorry for the delay. LGTM, thanks!

**Pull request merged**